tools/prow-auto-bumper/main.go

```diff
@@ -0,0 +1,282 @@
+		pv.m[image] = make([]versions, 0, 0)
+	}
+	_, variant := deconstructTag(tag)
+	iv := -1
+	for i, vs := range pv.m[image] {
+		if vs.variant == variant {
+			iv = i
+		}
+	}
+	if -1 == iv {
+		pv.m[image] = append(pv.m[image], versions{variant: variant})
+		iv = len(pv.m[image]) - 1
+	}
+	f(pv, iv)
+	return nil
+}
+
+func (pv *PRVersions) addNewTag(image, tag string) error {
+	return pv.addTag(
+		image,
+		tag,
+		func(pv *PRVersions, i int) {
+			pv.m[image][i].newVersion = tag
+		})
+}
+
+func (pv *PRVersions) addOldTag(image, tag string) error {
+	return pv.addTag(
+		image,
+		tag,
+		func(pv *PRVersions, i int) {
+			pv.m[image][i].oldVersion = tag
+		})
+}
+
+// Tags could be in the form of: v[YYYYMMDD]-[GIT_HASH](-[EXTRA_PART]),
+// separate it to `v[YYYYMMDD]-[GIT_HASH]` and `[EXTRA_PART]`
+func deconstructTag(in string) (string, string) {
+	dateCommit := in
+	var variant string
+	parts := strings.Split(in, "-")
+	if len(parts) > 2 {
+		variant = strings.Join(parts[2:], "-")
+	}
+	if len(parts) > 1 {
+		dateCommit = fmt.Sprintf("%s-%s", parts[0], parts[1])
+	}
+	return dateCommit, variant
+}
+
+// get key with highest value
+func getDominantKey(m map[string]int) string {
+	var res string
+	for key, v := range m {
+		if "" == res || v > m[res] {
+			res = key
+		}
+	}
+	return res
+}
+
+// The way k8s updates versions doesn't guaratee the same version tag across all iamges,
+// find the version that appears most times for reference
+func (pv *PRVersions) getDominantVersions() versions {
+	if nil != pv.v {
+		return *pv.v
+	}
+
+	cOld := make(map[string]int)
+	cNew := make(map[string]int)
+	for _, vss := range pv.m {
+		for _, vs := range vss {
+			normOldTag, _ := deconstructTag(vs.oldVersion)
+			normNewTag, _ := deconstructTag(vs.newVersion)
+			cOld[normOldTag]++
+			cNew[normNewTag]++
+		}
+	}
+
+	pv.v = &versions{
+		oldVersion: getDominantKey(cOld),
+		newVersion: getDominantKey(cNew),
+	}
+
+	return *pv.v
+}
+
+// parse changelist, find all version changes, and store them in image name: versions map
+func (pv *PRVersions) parseChangelist(ghc *ghutil.GithubClient) error {
+	fs, err := ghc.ListFiles(org, repo, *pv.PR.Number)
+	if nil != err {
+		return err
+	}
+	for _, f := range fs {
+		if nil == f.Patch {
+			continue
+		}
+		minuses := imageMinusRegexp.FindAllStringSubmatch(*f.Patch, -1)
+		for _, minus := range minuses {
+			pv.addOldTag(minus[imageImagePart], minus[imageTagPart])
+		}
+
+		pluses := imagePlusRegexp.FindAllStringSubmatch(*f.Patch, -1)
+		for _, plus := range pluses {
+			pv.addNewTag(plus[imageImagePart], plus[imageTagPart])
+		}
+	}
+
+	return nil
+}
+
+// Query all PRs from "k8s-ci-robot:autobump", find PR roughly 7 days old and was not reverted later.
+// Only return error if it's github related
+func getBestVersion(ghc *ghutil.GithubClient, org, repo string) (*PRVersions, error) {
+	visited := make(map[string]PRVersions)
+	var bestPv *PRVersions
+	var overallErr error
+	var bestDelta float64 = maxDelta + 1
```

Comment:
```suggestion
	bestDelta := maxDelta + 1
```

Reply:
The short variable declaration makes it an int, but `math.Abs(bestDelta)` near line 241 expects it to be float64, this is the best I can think of